src/query.c

```diff
@@ -348,8 +348,29 @@ typedef struct {
   size_t cap;
   QueryEvalCtx *q;
   QueryNodeOptions *opts;
+  double weight;
 } LexRangeCtx;
 
+static void rangeIterCbStrs(const char *r, size_t n, void *p, void *invidx) {
```

Comment:
Refactor out common parts; I can't tell the difference between the two functions.